rust/src/coreos_rootfs.rs

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2018 Red Hat, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ */
+
+//! # Code for CoreOS rootfs
+//!
+//! This backs the `rpm-ostree coreos-rootfs` CLI, which is intended as
+//! a hidden implmentation detail of CoreOS style systems.  The code
+//! is here in rpm-ostree as a convenience.
+
+use std::os::unix::prelude::*;
+use failure::{Fallible, ResultExt};
+use structopt::StructOpt;
+use openat;
+use nix;
+use libc;
+
+/// A reference to a *directory* file descriptor.  Unfortunately,
+/// the openat crate always uses O_PATH which doesn't support ioctl().
```

Comment:
And yeah, this turned out to be a lot more code in Rust than it would have been in C particularly if we just copy-pasted the [libostree code](https://github.com/ostreedev/ostree/blob/c6c640f3aefb4195432e77478f24bd4d842495e3/src/libostree/ostree-linuxfsutil.c) but I doggedly kept at it :wink: